File: interfaces/src/map.ts
export namespace Maps {

    /**
     * Represents an Node of an OsmWay
     *
     * @export
     * @interface IOsmNode
     * @extends {ICoords}
     */
    export interface IOsmNode extends ICoords {
        /**
         * An chunk is an field of coordinates to separate the map into pieces to make the indexing in the database easy.
         * 
         * The chunk are thr coordinates as a fixed-point number with 3 digits after the comma, concatenated from long and lat.
         *
         * @type {string}
         * @memberof IOsmNode
         */
        chunk: string;
        /**
         * The Id of the Node
         *
         * @type {string}
         * @memberof IOsmNode
         */
        id: string;
    }

    /**
     * Represents an path, which could be an way, building river or anything on the map
     *
     * @export
     * @interface IOsmWay
     */
    export interface IOsmWay {
        /**
         * The ids of the nodes of which the way consists.
         *
         * @see {@link IOsmNode}
         * @type {string[]}
         * @memberof IOsmWay
         */
        nodes: string[];
        /**
         * All chunks from which the nodes are inside.
         *
         * @type {string[]}
         * @memberof IOsmWay
         */
        chunks: string[];
        /**
         * The id of the way
         *
         * @type {string}
         * @memberof IOsmWay
         */
        id: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:level
         *
         * @type {string}
         * @memberof IOsmWay
         */
        level?: string;
        /**
         * Name of the represented object.
         *  
         * See: https://wiki.openstreetmap.org/wiki/Key:name
         *
         * @type {string}
         * @memberof IOsmWay
         */
        name?: string;
        /**
         * Is defined if the OsmWay is a street, a way or etc.
         * 
         * See: https://wiki.openstreetmap.org/wiki/Key:highway
         *
         * @type {string}
         * @memberof IOsmWay
         */
        highway?: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:access
         *
         * @type {string}
         * @memberof IOsmWay
         */
        access?: string;
        /**
         * An description of the object.
         * 
         * See: https://wiki.openstreetmap.org/wiki/Key:description
         *
         * @type {string}
         * @memberof IOsmWay
         */
        description?: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:surface
         *
         * @type {string}
         * @memberof IOsmWay
         */
        surface?: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:tracktype
         *
         * @type {string}
         * @memberof IOsmWay
         */
        tracktype?: string;
        /**
         * Is set if the osmWay represents an building
         * 
         * See: https://wiki.openstreetmap.org/wiki/Key:building
         *
         * @type {string}
         * @memberof IOsmWay
         */
        building?: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:amenity
         *
         * @type {string}
         * @memberof IOsmWay
         */
        amenity?: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:man_made
         *
         * @type {string}
         * @memberof IOsmWay
         */
        man_made?: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:waterway
         *
         * @type {string}
         * @memberof IOsmWay
         */
        waterway?: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:foot
         *
         * @type {string}
         * @memberof IOsmWay
         */
        foot?: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:layer
         *
         * @type {string}
         * @memberof IOsmWay
         */
        layer?: string;
        /**
         * See: https://wiki.openstreetmap.org/wiki/Key:tunnel
         *
         * @type {string}
         * @memberof IOsmWay
         */
        tunnel?: string;
    }

    /**
     * All Ways and nodes contained in a chunk
     *
     * @export
     * @interface IChunk
     */
    export interface IChunk {
        id: string;
        ways: IOsmWay[];
        nodes: IOsmNode[];
    }

    /**
     * Represents coordinates.
     *
     * @export
     * @interface ICoords
     */
    export interface ICoords {
        lon: number;
        lat: number;
    }

    /**
     * Represents an Point on the screen.
     *
     * @export
     * @interface IPoint
     */
    export interface IPoint {
        x: number;
        y: number;
    }

    export interface IViewport {
        height: number;
        width: number;
        scale: number;
        preRenderBorderWidth: number;
    }

    export type ISpeedDataState = 'initial' | 'pause' | 'walking' | 'running' | 'slow' | 'driving';

    export interface ISpeedData {
        groupId: string;
        timestamp: Date;
        state: ISpeedDataState;
        speed: number;
        position: ICoords;
    }

}

File: osm-import/models/map.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace osm_import.Map
{
    public class Node
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        [BsonElement("lat")]
        public double Lat { get; set; }
        [BsonElement("lon")]
        public double Lon { get; set; }
        [BsonElement("chunk")]
        public string Chunk { get; set; }
        [BsonElement("id")]
        public string OsmId { get; set; }
        [BsonElement("version")]
        public string Version { get; set; }
    }
    public class Way
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        [BsonElement("chunks")]
        public string[] Chunks { get; set; }
        [BsonElement("nodes")]
        public string[] Nodes { get; set; }
        [BsonElement("id")]
        public string OsmId { get; set; }
        [BsonElement("version")]
        public string Version { get; set; }
        [BsonElement("room")]
        public string Room { get; set; }
        [BsonElement("level")]
        public string Level { get; set; }
        [BsonElement("name")]
        public string Name { get; set; }
        [BsonElement("highway")]
        public string Highway { get; set; }
        [BsonElement("access")]
        public string Access { get; set; }
        [BsonElement("description")]
        public string Description { get; set; }
        [BsonElement("surface")]
        public string Surface { get; set; }
        [BsonElement("tracktype")]
        public string Tracktype { get; set; }
        [BsonElement("building")]
        public string Building { get; set; }
        [BsonElement("amenity")]
        public string Amenity { get; set; }
        [BsonElement("man_made")]
        public string Man_made { get; set; }
        [BsonElement("waterway")]
        public string Waterway { get; set; }
        [BsonElement("foot")]
        public string Foot { get; set; }
        [BsonElement("layer")]
        public string Layer { get; set; }
        [BsonElement("tunnel")]
        public string Tunnel { get; set; }
    }
}